Neos.Fusion/Classes/FusionObjects/Helpers/LazyProps.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Neos\Fusion\FusionObjects\Helpers;
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Fusion\Core\Runtime;
+
+/**
+ * @Flow\Proxy(false)
+ */
+final class LazyProps implements \ArrayAccess, \Iterator
```

Comment:
Does it make sense to add this as `LazyEvaluationProxy` to the Fusion/Core? That would allow to use it in other places aswell.

Reply:
I thought about it too, but this is rather specific for holding an array of keys and defering the evaluation of these. I'm not sure where and how we would re-use this. By only checking the interfaces in Runtime we always have the chance to extract a common class or something else later.

Reply:
:-) i now have a usecase for that. Evaluating only the current page of a multipage form. But since i the fusion-forms shall support Neos 4.3 i will copy that code anyways but thanks for the inspiration.